### ISSUE TITLE
Multiple, irrelevant GET Requests on elastic search with an infinite scroll.

### ISSUE THREAD

**Issue by therenix** [jhipster/generator-jhipster#3999]
##### **Overview of the issue**

When performing an elastic search on a view using infinite scroll for pagination, multiple GET requests are generated for the search term -- all for a different "page" of the result set, and none of which are necessarily even valid (we don't know if we're expecting 1 page, or 200 at this point because the search results haven't been returned yet.)

##### **Motivation for or Use Case**

It's generating unnecessary network traffic needlessly and probably taxing the server some unnecessarily.

##### **Reproduce the error**

In an application generated with Elastic Search enabled, generate a new entity and specify it to use infinite-scroll pagination.  Next, input 2 of the entities with random data.  Then return to the entity list, open up dev tools and watch the network after you perform a search.  A collection of get requests will go out for different sequential pages of the search results -- it will be a random number -- for us, it's sometimes 6 or 8; other times, it could be as much as 44.   Here's an example of me searching my Contracts entity for 'ABC123' (there are only two contracts in the system and yet it's requested 44 pages of results.)

![image](https://cloud.githubusercontent.com/assets/11657800/17755346/2b8370a4-649f-11e6-94f8-f1b135d3c136.png)

##### **Related issues**

I couldn't find any.

##### **Suggest a Fix**

The problem is ng-infinite-scroll is configured in the HTML like this:
`<tbody infinite-scroll="vm.loadPage(vm.page + 1)" infinite-scroll-disabled="vm.page >= vm.links['last']">`

So we're going to loadPages continuously until vm.page >= vm.links['last'], but the actual search method on the controller sets `vm.links = null` and doesn't give it a meaningful value until the onSuccess method when it actually gets the results from the server.   Meaning the client keeps executing the search query GET for page after page until the results come back.

Since we don't know how many pages to ask for until the results come back from the search, my suggestion would be to modify line 94 of generators/entity/templates/src/main/webapp/app/entities/infinite-scroll-template.ejs and replace it with:

```Javascript
        vm.links = {
            last: 0
        };
```

This results in a single request until the results come back, at which point, the links will be assigned the correct value and everything will go back to normal.

Note the same should probably be done on line 80 in the clear() method.

##### **JHipster Version(s)**
3.5.1

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**
```yaml
{
  "generator-jhipster": {
    "jhipsterVersion": "3.5.1",
    "baseName": "CCS_Tr",
    "packageName": "com.somevalue.ccstr",
    "packageFolder": "com/somevalue/ccstr",
    "serverPort": "8080",
    "authenticationType": "oauth2",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": "no",
    "websocket": "no",
    "databaseType": "sql",
    "devDatabaseType": "mysql",
    "prodDatabaseType": "mysql",
    "searchEngine": "elasticsearch",
    "buildTool": "gradle",
    "useSass": false,
    "applicationType": "monolith",
    "testFrameworks": [
      "gatling",
      "cucumber",
      "protractor"
    ],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "ja",
      "pt-br"
    ],
    "enableSocialSignIn": false
  }
}
```

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**
The only relevant piece of information is the pagination type:

```JSON
  "changelogDate": "20160803224502",
  "entityTableName": "contract",
  "dto": "no",
  "pagination": "infinite-scroll",
  "service": "serviceClass"
```

##### **Browsers and Operating System**
Chrome and Firefox on iOS (but probably all of them)

**Comment by deepu105** [comment]
would you like to do a PR to fix this?

**Comment by jdubois** [comment]
Yes @therenix I'm the one who coded both "infinite scroll" and the Elasticsearch part, but to be honest I didn't work too much on using both together.

- you seem to have a good solution, could you do a PR for that?
- @deepu105 this would need to be done on the ng2 branch too, can you help on that?

**Comment by deepu105** [comment]
this wouldn't affect ng2 yet, as the change is on entity sub generator we haven't migrated that yet and I'll do a merge from master before I create entity sub for ng2 anyway. so its ok

**Comment by jdubois** [comment]
Perfect then!

**Comment by therenix** [comment]
Thanks guys -- PR is done.

**Comment by deepu105** [comment]
closed via #4003
